grammar.js

```diff
@@ -878,13 +878,16 @@ module.exports = grammar(C, {
       '"',
     ),
 
-    subscript_expression: $ => prec(PREC.SUBSCRIPT, seq(
-      field('argument', $._expression),
+    subscript_argument_list: $ => seq(
       '[',
-      field('index', choice($._expression, $.initializer_list)),
+      commaSep(choice($._expression, $.initializer_list)),
       ']',
-    )),
+    ),
 
+    subscript_expression: $ => prec(PREC.SUBSCRIPT, seq(
+      field('argument', $._expression),
+      field('indices', $.subscript_argument_list),
+    )),
```

Comment:
I don't think there's a particular need to segment the argument list out into their own node - let's just keep it contained in `subscript_expression` instead. 

Otherwise this looks good.

So, just changing it to `commaSep(field('index', choice($._expression, $.initializer_list)))` would be good.

Reply:
OK - I  can do that easily enough. However the reason to present the argument list as its own node is to make it easier for higher-level code to treat it semantically like a list of function args. For example imagine an editor having a function to swap arguments around - not an uncommon operation when refactoring code.

I've written code to do this in my editor and it's useful to have a similar parent-node-child-node structure for:
 - function parameter lists
 - function argument lists
 - template parameter lists
 - template argument lists
 - subscript argument lists

Which was my rationale for doing it this way. Do you still want to change it back?

Reply:
Hm, you have a valid point, this is ok as is then

Can you just move the argument list below the expression then? Just a stylistic preference

Reply:
:+1: